File: Review-00-CPP-Head-to-Head-Testing/Example-1/even_gen.cpp
#include <iostream>

using namespace std;

int main(int argc, char** argv)
{
    // Check and parse command line args
    if (argc < 3) {
       cout << " Usage: ./even_gen [lower_bound] [upper_bound]" << "\n";
       return 1;
    }

    // Assume all args are well formed (i.e., can be parsed as integers).
    int lower_bound = atoi(argv[1]);
    int upper_bound = atoi(argv[2]);

    // The core even output logic
    cout << "Range [" << lower_bound << ", " << upper_bound << "]" << "\n";
    cout << "\n";

    for (int next_even = lower_bound; next_even <= upper_bound; next_even++) {
        cout << next_even << "\n";
    }

    return 0;
}

File: Review-00-Python-Head-to-Head-Testing/Example-1/even_gen.py
import sys


def main():
    # Check and parse command line args
    if len(sys.argv) < 3:
        print(" Usage: ./even_gen [lower_bound] [upper_bound]")
        sys.exit(1)

    # Assume all args are well formed (i.e., can be parsed as integers).
    lower_bound = int(sys.argv[1])
    upper_bound = int(sys.argv[2])

    # The core even output logic
    print(f"Range [{lower_bound}, {upper_bound}]")
    print()

    for next_even in range(lower_bound, upper_bound + 1):
        print(f"{next_even}")


if __name__ == "__main__":
    main()
